0-bootstrap/modules/jenkins/main.tf

```diff
@@ -0,0 +1,126 @@
+  source                      = "terraform-google-modules/project-factory/google"
+  version                     = "~> 7.0"
+  name                        = local.cicd_project_name
+  random_project_id           = true
+  disable_services_on_destroy = false
+  folder_id                   = var.folder_id
+  org_id                      = var.org_id
+  billing_account             = var.billing_account
+  activate_apis               = local.activate_apis
+  labels                      = var.project_labels
+}
+
+resource "google_project_service" "jenkins_apis" {
+  for_each           = toset(local.jenkins_apis)
+  project            = module.cicd_project.project_id
+  service            = each.value
+  disable_on_destroy = false
+}
+
+/******************************************
+  Jenkins Agent GCE instance
+*******************************************/
+resource "google_service_account" "jenkins_agent_gce_sa" {
+  project      = module.cicd_project.project_id
+  account_id   = var.jenkins_sa_email
+  display_name = "CFT Jenkins Agent GCE custom Service Account"
+}
+
+resource "google_compute_instance" "jenkins_agent_gce_instance" {
+  project      = module.cicd_project.project_id
+  name         = var.jenkins_agent_gce_name
+  machine_type = "n1-standard-1"
+  zone         = "europe-west2-a"
+
+  tags = local.jenkins_gce_fw_tags
+
+  boot_disk {
+    initialize_params {
+      image = "debian-cloud/debian-9"
+    }
+  }
+
+  network_interface {
+    network = google_compute_network.jenkins_agents.self_link
+
+    access_config {
+      // Ephemeral IP
+      // TODO(caleonardo): This should have either a fixed IP address or no external IP at all
+    }
+  }
+
+  // Adding ssh public keys to the metadata, so the Jenkins Master can connect
+  metadata = {
+    enable-oslogin = "false"
+    ssh-keys = "${var.jenkins_agent_gce_ssh_user}:${file(var.jenkins_agent_gce_ssh_pub_key_file)}"
+  }
+
+  // TODO(caleonardo): Setup the Java installation here for the Jenkins Agent
+  metadata_startup_script = "echo hi > /test.txt"
+
+  service_account {
+    email = google_service_account.jenkins_agent_gce_sa.email
+    scopes = [
+      "https://www.googleapis.com/auth/compute.readonly",
+      "https://www.googleapis.com/auth/devstorage.read_only",
+      "https://www.googleapis.com/auth/userinfo.email",
+    ]
+  }
+
+  // allow stopping the GCE instance to update some of its values
+  allow_stopping_for_update = true
+}
+
+/******************************************
+  Jenkins Agent GCE Firewall rules
+*******************************************/
+
+resource "google_compute_firewall" "allow_ssh_to_jenkins_agent_fw" {
+  project = module.cicd_project.project_id
+  name    = "allow-ssh-to-jenkins-agents"
+  description = "Allow the Jenkins Master (Client) to connect to the Jenkins Agents (Servers) using SSH."
+  network = google_compute_network.jenkins_agents.name
+
+  source_ranges = ["0.0.0.0/0"]  // TODO(caleonardo): Specify here the Jenkins Master IP Address
```

Comment:
We can just parameterize this and make this a required value.

Reply:
Done.